nemo/collections/common/tokenizers/bytelevel_tokenizers.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from pathlib import Path
+from typing import List
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+
+__all__ = ['ByteLevelProcessor', 'ByteLevelTokenizer']
+
+
+class ByteLevelProcessor:
+    """
+    A very basic tokenization and detokenization class for use with byte-level
+    tokenization.
+    """
+
+    def detokenize(self, tokens: List[str]) -> str:
+        return ' '.join(tokens)
```

Comment:
A small question, why are we adding whitespace while detokenizing while we are simply returning the text at the time of tokenization?